### DISCUSSION TITLE
Big right margin in PDF version

### DISCUSSION THREAD

**Discussion by rsbyrne** [executablebooks/meta#358] [Q&A] [answered]
Hi all,

I'm a new (and enthusiastic!) user of JupyterBooks. I'm currently using it to publish my thesis online:
https://rsbyrne.github.io/thesis

My question is about how to publish my JupyterBook as a single PDF which looks 'traditional' enough to hand in to my supervisors as a finished product.

Specifically, when using:
`jb build --builder pdfhtml`

The output by default includes the table of contents from the HTML version. I'm aware that this table can be hidden using `css` as documented in: https://jupyterbook.org/advanced/pdf.html

However, even after rendering the table invisible, enlarged right margins seem to persist in the place where the table would usually go. I'm wondering if anyone knows the proper way to remove this extra margin space, and otherwise make a Jupyter Book exportable to PDF in a way that makes it suitable for printing and handing in to the board of examiners.

Thanks in advance,
Rohan

**Comment by jstac** [comment, marked as answer]
@AakashGfude and @mmcky have been working on this, via latex.

You can see the results in the "book" PDF here: https://python-programming.quantecon.org/intro.html

Is that the direction that you want to go in?

PS Nice title for your thesis.

**Reply by rsbyrne** [reply]
Oh, thanks so much!

That example is absolutely perfect - exactly what I'm going for.

@AakashGfude @mmcky if you don't mind, I might have a poke around your awesome work and see if I can get an understanding of how you did it, and perhaps borrow some of your styles with grateful attribution. I also hope it's okay if, along the way, I get in touch with a couple of (hopefully well-informed) queries.

In the future, I reckon just about every scientific document will be published this way. We've just got to break the trail!

Cheers all.

**Reply by mmcky** [reply]
@rsbyrne if you are running the latest version of `jupyter-book` you should be able to build via LaTeX now using:

```
jb build <project> --builder=pdflatex
```

this should use [jupyterbook-latex](https://github.com/executablebooks/jupyterbook-latex) which is a `jupyter-book` extension for building book style `pdf` files for projects. Let me know how you go -- happy to help. 

Everything in the `intro` (or first document) is treated as `frontmatter` and the rest of the document is `chapter` content or `part`/`chapter` content depending on how your `_toc.yml` is structured.

**Reply by rsbyrne** [reply]
Yup @mmcky - you were quite right: I was using `--builder pdfhtml` instead of `pdflatex`. Now using `pdflatex` and it's looking much closer to what you've got in your gorgeous quantecon textbook.

I assume the rest is a matter of carefully tuning various configs and css sheets to get the ideal look - still a bit of a learning curve for me. But at least I can see now that JupyterBook definitely *can* do everything I need it to.

Thanks folks for helping out a newbie. I'll post again when I hit another hurdle - and I'll make sure to share the finished product too when I get there (with plenty of shoutouts in the acknowledgements!)
